lightning/src/routing/router.rs

```diff
@@ -375,10 +375,13 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
 		return Err(LightningError{err: "Cannot send a payment of 0 msat".to_owned(), action: ErrorAction::IgnoreError});
 	}
 
-	let last_hops = last_hops.iter().filter_map(|hops| hops.0.last()).collect::<Vec<_>>();
-	for last_hop in last_hops.iter() {
-		if last_hop.src_node_id == *payee {
-			return Err(LightningError{err: "Last hop cannot have a payee as a source.".to_owned(), action: ErrorAction::IgnoreError});
+	for route in last_hops.iter() {
+		if route.0.len() > 0 {
```

Comment:
The code was raising index out of bounds errors for that. This check guards against that.

Reply:
As of commit [0c09852](https://github.com/rust-bitcoin/rust-lightning/pull/1040/commits/0c09852311b3b7c42d1210aa69aa3f827a91655d) this became redundant because I used `.filter(|route| !route.0.is_empty())` as advised.